### PR TITLE
Add urlFieldsFilter option to cleanUrlTracker

### PR DESCRIPTION
**Pull request by philipwalton**
/cc @idris @pbowyer, please let me know if you have any feedback on the [new options](https://github.com/googleanalytics/autotrack/blob/13b94ec5be433b5979dfb4888f4294625884f997/docs/plugins/clean-url-tracker.md)

### CODE REVIEW

**Comment by idris** [comment]
Looks like that should work...

I'd maybe change the example: https://github.com/googleanalytics/autotrack/blob/13b94ec5be433b5979dfb4888f4294625884f997/docs/plugins/clean-url-tracker.md#using-the-urlfieldsfilter-option

Right now the example changes the `location`, which you warn against in documenting the `urlFieldsFilter` option.

Perhaps a better example would be to remove ids from a URL like `https://example.com/projects/1234/tasks`

**Comment by philipwalton** [comment]
@idris thanks, good idea!

**Comment by pbowyer** [comment]
Excellent work @philipwalton!

I've read through the code and documentation and it looks good. Didn't understand why `requireCleanUrlTracker_urlFieldsFilter()` and `requireCleanUrlTracker_multipleOpts()` were needed, but I've never used this testing framework.

File: lib/plugins/max-scroll-tracker.js [hunk 1]
```diff
@@ -202,8 +202,9 @@ class MaxScrollTracker {
    * @return {number}
    */
   getPagePath() {
-    return parseUrl(
-        this.tracker.get('page') || this.tracker.get('location')).path;
+    const url = parseUrl(
```
Comment (pbowyer) [inline lib/plugins/max-scroll-tracker.js]: Does this change (L205-207) belong in this PR? The rest are about the clean URL plugin, while this is the max-scroll-tracker

Comment (philipwalton) [inline lib/plugins/max-scroll-tracker.js]: Yeah, I think this is just a Github error. It's not showing the correct base branch for some reason. When I merge it won't be in the history.

**Comment by pbowyer** [comment]
Thank you!

**Pull request merged**